source/module_base/module_mixing/test/mixing_test.cpp

```diff
@@ -0,0 +1,268 @@
+    {
+        if (method == "broyden")
+        {
+            this->mixing = new Base_Mixing::Broyden_Mixing(this->mixing_ndim, this->mixing_beta);
+        }
+        else if (method == "pulay")
+        {
+            this->mixing = new Base_Mixing::Pulay_Mixing(this->mixing_ndim, this->mixing_beta);
+        }
+        else if (method == "plain")
+        {
+            this->mixing = new Base_Mixing::Plain_Mixing(this->mixing_beta);
+        }
+    }
+
+    void clear()
+    {
+        delete this->mixing;
+        this->mixing = nullptr;
+    }
+
+    /**
+     * @brief sover linear equation:
+     *        [ 8 -3  2 ][x1]   [20]       [3]
+     *        [ 4 11 -1 ][x2] = [33]   x = [2]
+     *        [ 6  3 12 ][x3]   [36]       [1]
+     *
+     *         [x1]   [ 3/8  -2/8   20/8 ][x1]
+     *         [x2] = [-4/11  1/11 -33/11][x2]
+     *         [x3]   [-6/12 -3/12  36/12][x3]
+     */
+    template <typename FPTYPE>
+    void solve_linear_eq(FPTYPE* x_in, FPTYPE* x_out)
+    {
+        this->mixing->init_mixing_data(xdata, 3, sizeof(FPTYPE));
+        std::vector<FPTYPE> delta_x(3);
+
+        auto screen = std::bind(&Mixing_Test::Kerker_mock<FPTYPE>, this, std::placeholders::_1);
+        auto inner_product
+            = std::bind(static_cast<double (Mixing_Test::*)(FPTYPE*, FPTYPE*)>(&Mixing_Test::inner_product_mock),
+                        this,
+                        std::placeholders::_1,
+                        std::placeholders::_2);
+
+        double residual = 10.;
+        this->niter = 0;
+        while (niter < maxiter)
+        {
+            x_out[0] = (3. * x_in[1] - 2. * x_in[2] + 20.) / 8.;
+            x_out[1] = (-4. * x_out[0] + 1. * x_in[2] + 33.) / 11.;
+            x_out[2] = (-6. * x_out[0] - 3. * x_out[1] + 36.) / 12.;
+
+            niter++;
+
+            for (int i = 0; i < 3; ++i)
+            {
+                delta_x[i] = x_out[i] - x_in[i];
+            }
+            residual = this->inner_product_mock(delta_x.data(), delta_x.data());
+            if (residual <= thr)
+            {
+                break;
+            }
+
+            this->mixing->push_data(this->xdata, x_in, x_out, screen, true);
+
+            this->mixing->cal_coef(this->xdata, inner_product);
+
+            this->mixing->mix_data(this->xdata, x_in);
+        }
+    }
+
+    template <typename FPTYPE>
+    void Kerker_mock(FPTYPE* drho)
+    {
+    }
+
+    double inner_product_mock(double* x1, double* x2)
+    {
+        double xnorm = 0.0;
+        for (int ir = 0; ir < 3; ++ir)
+        {
+            xnorm += x1[ir] * x2[ir];
+        }
+        return xnorm;
+    }
+    double inner_product_mock(std::complex<double>* x1, std::complex<double>* x2)
+    {
+        double xnorm = 0.0;
+        for (int ir = 0; ir < 3; ++ir)
+        {
+            xnorm += x1[ir].real() * x2[ir].real() + x1[ir].imag() * x2[ir].imag();
+        }
+        return xnorm;
+    }
+};
+
+TEST_F(Mixing_Test, Broyden_Solve_LinearEq)
```

Comment:
```suggestion
TEST_F(MixingTest, BroydenSolveLinearEq)
```
underline "_" might be discouraged in unit test, if I remember correctly :(
If so, it is better to change names of test suite and tests, thanks

Reply:
Ok